state/lease/client.go

```diff
@@ -360,36 +275,24 @@ func (client *client) extendLeaseOps(name string, request lease.Request) ([]txn.
 		return nil, entry{}, lease.ErrInvalid
 	}
 
-	// According to the local clock, we want the lease to extend until
-	// <duration> in the future.
-	now := client.config.Clock.Now()
-	expiry := now.Add(request.Duration)
-
-	// We don't know what time the original writer thinks it is, but we
-	// can figure out the earliest and latest local times at which it
-	// could be expecting its original lease to expire.
-	skew := client.skews[lastEntry.writer]
-	if expiry.Before(skew.Earliest(lastEntry.expiry)) {
-		// The "extended" lease will certainly expire before the
-		// existing lease could. Done.
-		return nil, lastEntry, errNoExtension
+	globalTime, err := client.refreshGlobalTime()
```

Comment:
Originally that was my plan, but that would be unsound. If we were to do that, then claims could last less than the specified duration, because the reader might lag arbitrarily long behind the updater.